src/dms/core/EdFi.DataManagementService.Core.Tests.Unit/Middleware/JwtRoleAuthenticationMiddlewareTests.cs

```diff
@@ -0,0 +1,490 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Security.Claims;
+using EdFi.DataManagementService.Core.External.Frontend;
+using EdFi.DataManagementService.Core.External.Model;
+using EdFi.DataManagementService.Core.Middleware;
+using EdFi.DataManagementService.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+using EdFi.DataManagementService.Core.Security;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using NUnit.Framework;
+
+namespace EdFi.DataManagementService.Core.Tests.Unit.Middleware;
+
+[TestFixture]
+[Parallelizable]
+public class JwtRoleAuthenticationMiddlewareTests
```

Comment:
A companion to the other JWT Auth middleware, this one is a bit speculative and is for endpoints that we want to secure but don't use ClientAuthorizations (like a management endpoint or something)